jerry-ext/include/jerryscript-ext/autorelease.impl.h

```diff
@@ -0,0 +1,33 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_AUTORELEASE_IMPL_H
+#define JERRYX_AUTORELEASE_IMPL_H
+
+#include "jerryscript.h"
+
+#ifdef __GNUC__
+/*
+ * Calls jerry_release_value (*value).
+ * The GCC __cleanup__ function must take a pointer to the variable to clean up.
+ */
+void jerryx_autorelease_cleanup (const jerry_value_t *value);
```

Comment:
Yeah, that's the way `__cleanup__` works:

> The function must take one parameter, a pointer to a type compatible with the variable.

https://gcc.gnu.org/onlinedocs/gcc/Common-Variable-Attributes.html